practice/tasks/IDisposable/IDisposableImplementation/IDisposableImplementation/MemoryStreamLogger.cs

```diff
@@ -1,22 +1,52 @@
-﻿using System.IO;
+﻿using System;
+using System.IO;
 
 namespace NetMentoring
 {
-    public class MemoryStreamLogger
+    public class MemoryStreamLogger : IDisposable
     {
-        private FileStream memoryStream;
-        private StreamWriter streamWriter;
+        private StreamWriter _streamWriter;
+        private StreamReader _streamReader;
+        private const string PathToFile = @"Files\log.txt";
+        private bool _disposed;
 
-        public MemoryStreamLogger()
+        public void ReadFile()
         {
-            memoryStream = new FileStream(@"\log.txt", FileMode.OpenOrCreate);
-            streamWriter = new StreamWriter(memoryStream);
+            _streamReader = new StreamReader(PathToFile);
+            var counter = 0;
+            string line;
+            while ((line = _streamReader.ReadLine()) != null)
```

Comment:
we are already in using, so it will handle the error
also, we can add one thy catch block to check IOException/OutOfMemoryException and close it in finally

Reply:
we will not be able to read file?